src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -145,36 +116,18 @@
     public Path[] locations();
 
     /**
-     * Returns the translog file with the given id as a Path. This
-     * will return a relative path.
-     */
-    Path getPath(long translogId);
-
-    /**
      * return stats
      */
     TranslogStats stats();
 
     /**
-     * Returns the largest translog id present in all locations or <tt>-1</tt> if no translog is present.
+     * notifies the translog that translogId was committed as part of the commit data in lucene, together
+     * with all operations from previous translogs. This allows releasing all previous translogs.
+     *
+     * @throws FileNotFoundException if the given translog id can not be found.
      */
-    long findLargestPresentTranslogId() throws IOException;
+    void committedTranslogId(long translogId) throws FileNotFoundException;
```

Comment:
I like the mark comitted  name. renamed. The exception is very specific as it is thrown by the code it self, not an IO layer.